### PR TITLE
[Merged by Bors] - Basic documentation for Entities, Components and Systems

### PR DESCRIPTION
**Pull request by alice-i-cecile**
These are largely targeted at beginners, as `Entity`, `Component` and `System` are the most obvious terms to search when first getting introduced to Bevy.

### CODE REVIEW

File: crates/bevy_ecs/src/system/system.rs [hunk 1]
```diff
@@ -17,6 +17,11 @@ impl SystemId {
 }
 
 /// An ECS system that can be added to a [Schedule](crate::schedule::Schedule)
+///
+/// Systems are just ordinary Rust functions, each of whose parameters implement the [IntoSystem] trait.
```
Comment (MinerSebas) [inline crates/bevy_ecs/src/system/system.rs:21]:
```suggestion
/// Systems are just ordinary Rust functions, where each parameter implements the [SystemParam] trait.
```
This flows much better, and references the correct Trait.

File: crates/bevy_ecs/src/component/mod.rs [hunk 2]
```diff
@@ -11,6 +11,19 @@ use std::{
...
+/// Any Rust object, from a raw data type to an enum to a custom struct can be used as a `Component` without any extra boilerplate
+/// if and only if it fulfills the `Send + Sync + `static` bounds.
+///
+/// Components are commonly added to entities using [Commands::spawn](crate::system::Commands::spawn),
+/// [Commands::insert](crate::system::Commands::insert) or the [World](crate::world::World) equivalents.
+///
+/// You can query for components in your systems by including a [Query](crate::system::Query)-type function argument in your system,
+/// which will be automatically fetched from the [World](crate::world::World) by the scheduler.
+///
+/// A collection of components is known as a [Bundle](crate::bundle::Bundle).
```
Comment (Ratysz) [inline crates/bevy_ecs/src/component/mod.rs:26]:
```suggestion
/// A component is data associated with an `Entity`. Each entity can have multiple different types
/// of components, but only one of them per type.
///
/// Any type that is `Send + Sync + 'static` automatically implements `Component`.
///
/// Components are added with new entities using [Commands::spawn](crate::system::Commands::spawn),
/// or to existing entities with [Commands::insert](crate::system::Commands::insert),
/// or their [World](crate::world::World) equivalents.
///
/// Components can be accessed in systems by using a [Query](crate::system::Query)
/// as one of the arguments.
///
/// Components can be grouped together into a [Bundle](crate::bundle::Bundle).
```

File: crates/bevy_ecs/src/entity/mod.rs [hunk 3]
```diff
@@ -13,7 +13,10 @@ use std::{
 
 /// Lightweight unique ID of an entity
 ///
-/// Obtained from `World::spawn`. Can be stored to refer to an entity in the future.
+/// Obtained from [World::spawn](crate::world::World::spawn), typically via [Commands::spawn](crate::system::Commands::spawn). Can be stored to refer to an entity in the future.
+///
+/// You can get the `Entity` ID of entities returned in your queries by using `Query<(Entity, &MyComponent)>`.
+/// Use [Query::get](crate::system::Query::get) and related methods to extract the components of a specific entity.
```
Comment (Ratysz) [inline crates/bevy_ecs/src/entity/mod.rs:19]:
```suggestion
/// `Entity` can be a part of a query, e.g. `Query<(Entity, &MyComponent)>`.
/// Components of a specific entity can be accessed using
/// [Query::get](crate::system::Query::get) and related methods.
```

File: crates/bevy_ecs/src/system/system.rs [hunk 4]
```diff
@@ -17,6 +17,11 @@ impl SystemId {
 }
 
 /// An ECS system that can be added to a [Schedule](crate::schedule::Schedule)
+///
+/// Systems are just ordinary Rust functions, where each parameter implements the [SystemParam] trait.
+/// You can add systems to your app by including `.add_system(my_system.system())` in your `AppBuilder`.
+/// In Bevy apps, the scheduler will automatically run systems in parallel, coordinate resource access
+/// and call your systems each pass of the main loop.
```
Comment (Ratysz) [inline crates/bevy_ecs/src/system/system.rs:24]:
```suggestion
/// Systems are functions with all arguments implementing [SystemParam].
///
/// Systems are added to an application using `AppBuilder::add_system(my_system.system())`
/// or similar methods, and will run once per pass of the main loop.
///
/// Systems are executed in parallel, in opportunistic order; data access is managed automatically.
/// It's possible to specify explicit execution order between specific systems,
/// see [SystemDescriptor](crate::schedule::SystemDescriptor).
```

**Review by Ratysz — COMMENTED** [review]
I've got a bit carried away during reveiw. You don't have to agree with any of these changes; I'll list my reasons for them anyway.

`Component` doc reads as subjective: "in a straightforward and efficient fashion", "is very simple", "commonly", "is known as"...

Components are not what stores - they are what is stored.

I understand the desire to use "if and only if", but "`X` is `Trait + OtherTrait`" is unambiguous, more popular in Rust docs, and allows for some seriously concise sentences.

It's not necessary to mention Rust. The users do, presumably, know what language they're working with.

I'm partial to dry and impersonal style - this is technical documentation, it should be as to the point as possible. It does have the unfortunate tendency to start all sentences the same way, though.

Trimmed lines to 100 symbols - this is what `rustfmt` usually targets for code; maintaining that for comments and docs is not required, but I like it for consistency (and it removes the horizontal scrollbar in my IDE).

**Comment by alice-i-cecile** [comment]
I liked your changes a lot @Ratysz, and you're likely correct about tone, even if it's a bit drier than is perhaps ideal for certain audiences. We can save a more approachable tone for examples and the book.

 Very good idea to include a link to the docs on how to order systems in the `System` docs.
 
 I had expected `rustfmt` to trim the doc comments to the appropriate length automatically 😅

File: crates/bevy_ecs/src/system/system.rs [hunk 5]
```diff
@@ -17,6 +17,15 @@ impl SystemId {
 }
 
 /// An ECS system that can be added to a [Schedule](crate::schedule::Schedule)
+///
+/// Systems are functions with all arguments implementing [SystemParam].
+///
+/// Systems are added to an application using `AppBuilder::add_system(my_system.system())`
+/// or similar methods, and will run once per pass of the main loop.
```
Comment (Ratysz) [inline crates/bevy_ecs/src/system/system.rs:24]:
```suggestion
/// or similar methods, and will generally run once per pass of the main loop.
```
I was trying to find balance between brevity and detailing out the exceptions and instead failed to bring them up at all 🤦 
I don't have a better solution.

**Review by Ratysz — APPROVED** [review]

File: crates/bevy_ecs/src/system/system.rs (hunk 5 above)
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/system/system.rs:24]: I agree: this is a good compromise in these docs for now.

**Comment by cart** [comment]
Looks good to me!

**Pull request closed without merging**